src/site/content/en/accessible/control-focus-with-tabindex/index.md

```diff
@@ -142,6 +142,113 @@ You can learn more about them in our guide on
 [screen reader basics](/semantics-and-screen-readers).
 {% endAside %}
 
+{% AssessmentCallout 'Use the drop-down below each code sample to check your understanding of tab order.' %}
+{% Tabs 'Samples for knowledge self check' %}
+{% Tab 'sample' %}
+
+This HTML renders a modal dialog:
+```html
+<div role="dialog" aria-labelledby="dialog-header">
+  <button aria-label="Close"></button>
+  <h2 id="dialog-header">
+    Do you want to allow notifications from this website?
+  </h2>
+  <button>No</button>
+  <button>Yes</button>
+</div>
+```
+
+{% AssessmentHint 'What is the tab order for the elements in the sample?' %}
+1. The **Close** button
+1. The **No** button
+1. The **Yes** button
+
+Only the `<button>` elements are included in the tab order
+because they're the only native HTML form elements.
+To insert other elements into the tab order, you would add a `tabindex` attribute.
+{% endAssessmentHint %}
+
+{% endTab %}
+{% Tab 'sample' %}
+
+```html
+<section tabindex="-1">
+  <h2>Cat facts</h2>
+  <ul>
+    <li>A group of cats is called a <a href="https://m-w.com/dictionary/clowder">clowder</a>.</li>
+    <li>Most cats are <a href="https://www.catfacts.org/catnip.html"> unaffected by catnip</a>.</li>
+  </ul>
+</section>
+```
+
+{% AssessmentHint 'Which elements from the sample are included in the tab order?' %}
+Only the `<a>` elements are included in the tab order.
```

Comment:
I thought the answer would be "none of them" because the tabindex of the `section` was `-1` and the `li` tags were children of the `section`. I did quick scan of the guide and I don't believe this idea is covered.

Self-assessments already proving their ability to really enhance knowledge, v exciting!

Which brings up an interesting topic. How do we capture the various incorrect answers? Ideally I think our solution text should try to explain each possible answer, and why it was wrong. But how do we solicit those incorrect answers from our users?

cc @robdodson @mfriesenhahn @devnook

Reply:
1st thing: This is actually an important issue. These are comprehension questions, so we shouldn't ask about concepts that aren't explicit in the parent article. @robdodson, what would you think about adding a para about this issue and maybe `inert`? Either that or I need to change the item.

2nd thing: Once we add multiple-choice formats, you're right, we'll need to write rationales for all options. I don't think we need to _solicit_ all possible responses from users though. The way one usually writes distractors (i.e., wrong answers) is to provide common misconceptions or other cognitive errors. If we think a correct response to a single item isn't likely to capture all the nuances of a particular issue, we should generally write another item. (I've tried to do that in these sets--for example, by asking about both same-domain and cross-domain links in the links-vs-buttons set.) To see how other people are handling rationales, check out [Khan Academy](https://www.khanacademy.org/) and/or [IXL](https://www.ixl.com/).

Reply:
Talked to @robdodson about issue #1 and will add a para in a separate PR.